src/components/EditSnack/EditSnack.js

```diff
@@ -1,26 +1,55 @@
-import React, {Component} from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import ButtonList from '../ButtonList/ButtonList';
+import ListSelection from '../listSelection/ListSelection';
 import Logo from '../Logo/Logo';
+import './EditSnack.css';
 
-class EditSnack extends Component {
-	handleClick = (id, name) => {
-		this.props.setActualItem(id);
-		this.props.history.push('/slackname');
-	};
+const importImages = require.context('./assets', true, /\.svg$/);
+const imgFilesObject = importImages.keys().reduce((images, key) => {
+	images[key] = importImages(key);
+	return images;
+}, {});
 
-	render() {
-		return (
-			<div>
-				<Logo />
-				<ButtonList
-					items={this.props.items}
-					onClick={this.handleClick}
-				/>
+const EditSnack = props => {
+	const items = props.items.map(item => {
+		const relativeImagePath = './' + item.image;
+		let actualImagePath;
+
+		if (imgFilesObject[relativeImagePath]) {
```

Comment:
I think it'd be good to extract this logic to a separate method to make it a bit more self describing, mostly my fault as it was just hacked together to get it working initially.

We could extract a constant for what image to use if none was provided (misc-bar.svg)

We could then pass in the function reference which is pretty nifty:

```
const addItemImage => item => {
  const imagePathWithFileType = item.image.endsWith('.svg') ? item.image : item.image + '.svg'
  const relativeImagePath = `./${imagePathWithFileType}`;

  const doesAssetExistForItem = imgFilesObject[relativeImagePath] !== undefined;
  const genericImagePath = './misc-bar.svg';

  const actualImagePath = doesAssetExistForItem ? relativeImagePath : genericImagePath;

  return {
    ...item,
    image: importImages(actualImagePath),
  };
}

const EditSnack = props => {
  const items = props.items.map(addItemImage)
  // ...
}

```

This way the variables are expressing what's happening, and it's really easy to break this further down if we wanted